utils/storage/storage.js

```diff
@@ -248,10 +251,42 @@ class Storage {
     );
   }
 
+  /**
+   * Function to fetch anonymousId from external source
+   * @param {string} key source of the anonymousId
+   * @returns string
+   */
+  fetchExternalAnonymousId(key) {
+    let anonId;
+
+    if (Object.keys(anonymousIdKeyMap).includes(key)) {
+      return anonId;
+    }
+    switch (key) {
+      case "segment":
+        if (Store.enabled) {
+          anonId = Store.get(anonymousIdKeyMap[key]);
+        }
+        if (!anonId && Cookie.IsCookieSupported()) {
+          anonId = Cookie.get(anonymousIdKeyMap[key]);
+        }
+        return anonId;
+
+      default:
+        return anonId;
+    }
+  }
+
   /**
    * get stored anonymous id
    */
-  getAnonymousId() {
+  getAnonymousId(anonymousIdOptions) {
+    const source = get(anonymousIdOptions, "source");
+    if (get(anonymousIdOptions, "enabled") && typeof source === "string") {
+      const anonId = this.fetchExternalAnonymousId(source.toLowerCase());
```

Comment:
Let's leave the `toLowerCase` conversion to the fetch method.

Reply:
done